circleguard/draw.py

```diff
@@ -78,10 +86,26 @@ def init():
         def update(i):
             plot1.set_data(data1[0][i - 100:i], data1[1][i - 100:i])
             plot2.set_data(data2[0][i - 100:i], data2[1][i - 100:i])
+            if i != 0 and i % int((total/10)) == 0:
+                print(f"Saved {round(100 * float(i)/float(total))}% already!")
             return plot1, plot2
 
-        animation = FuncAnimation(fig, update, frames=itr.count(100), init_func=init, blit=True, interval=dt)
+        setup()
+        
         plt.show(block=True)
+        plt.close('all')  # Maybe unnecessary
+        if input("Do you want to save the visualization? WARNING! This may take long ") == "y":
```

Comment:
`if input("Do you want to save the video? WARNING, this may take a while ")[0] == "y":`

Reply:
Inserting [0] indeed works, but this will fail if there is no input. I instead choose to use:
`if (answer and answer[0].lower().strip() == "y")`
This is also used in "comparer.py"